### PR TITLE
adding the solutions for 'reverse linkedlist'

### PR DESCRIPTION
**Pull request by ntl2105**
you know it!

### CODE REVIEW

File: reverse_linkedlist/jade_solution.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+def get_data_list(data):
+  data_list = []
+  data_list.append(one['data'])
```
Comment (cranthony) [inline reverse_linkedlist/jade_solution.py]: You can initialize `inner_dict` to `one`, and simply start the loop
```suggestion
  inner_dict = one
  while inner_dict is not None:
```

File: reverse_linkedlist/jade_solution.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  inner_dict = one['next']
```
Comment (cranthony) [inline reverse_linkedlist/jade_solution.py]: What do you think about naming this `current_node`?  I'm not sure how to interpret `inner_dict` when I read it.

File: reverse_linkedlist/jade_solution.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  while inner_dict is not None:    
+  #     inner_dict = one['next']
+      element_data = inner_dict['data']
+      data_list.append(element_data)
```
Comment (cranthony) [inline reverse_linkedlist/jade_solution.py]: You can combine this and the above line if you want

```suggestion
      data_list.append(inner_dict['data'])
```

File: reverse_linkedlist/jade_solution.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      inner_dict = inner_dict['next']
+  return data_list
+
+def make_dict(current_data, prev_dict):
```
Comment (cranthony) [inline reverse_linkedlist/jade_solution.py:11]: How about naming this `make_node`?

File: reverse_linkedlist/jade_solution.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    return {
+        'data':current_data,
+        'next': prev_dict
```
Comment (cranthony) [inline reverse_linkedlist/jade_solution.py:14]: Should `prev_dict` be named `next`?

File: reverse_linkedlist/jade_solution.py (hunk 2 above)
Comment (ntl2105) [reply reverse_linkedlist/jade_solution.py]: no!

File: reverse_linkedlist/jade_solution.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+def reverse_linked_list(data):
+  data_list = get_data_list(data)
+  for i,e in enumerate(data_list):
```
Comment (cranthony) [inline reverse_linkedlist/jade_solution.py:19]: Could you name these something clearer than `i` and `e`?

File: reverse_linkedlist/jade_solution.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      if not i:
```
Comment (cranthony) [inline reverse_linkedlist/jade_solution.py:20]: Does this ever happen?

**Review by cranthony — COMMENTED** [review]
Good work!

File: reverse_linkedlist/jade_solution.py (hunk 4 above)
Comment (ntl2105) [reply reverse_linkedlist/jade_solution.py:11]: got it!

File: reverse_linkedlist/jade_solution.py (hunk 6 above)
Comment (ntl2105) [reply reverse_linkedlist/jade_solution.py:19]: okay, fine.
next time.

**Pull request merged**